hibernate-core/src/main/java/org/hibernate/type/JaxbXmlFormatMapper.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.type;
+
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+
+import org.hibernate.internal.util.ReflectHelper;
+import org.hibernate.type.descriptor.WrapperOptions;
+import org.hibernate.type.descriptor.java.JavaType;
+
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Marshaller;
+import jakarta.xml.bind.Unmarshaller;
+import jakarta.xml.bind.annotation.XmlAnyElement;
+import jakarta.xml.bind.annotation.XmlRootElement;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+
+/**
+ * @author Christian Beikov
+ */
+public class JaxbXmlFormatMapper implements FormatMapper {
+
+	public static final String SHORT_NAME = "jaxb";
+	public static final JaxbXmlFormatMapper INSTANCE = new JaxbXmlFormatMapper();
+
+	public JaxbXmlFormatMapper() {
+	}
+
+	@Override
+	public <T> T fromString(CharSequence charSequence, JavaType<T> javaType, WrapperOptions wrapperOptions) {
+		try {
+			if ( Map.class.isAssignableFrom( javaType.getJavaTypeClass() ) ) {
+				final JAXBContext context;
+				final Class<Object> keyClass;
+				final Class<Object> valueClass;
+				if ( javaType.getJavaType() instanceof ParameterizedType ) {
+					final Type[] typeArguments = ( (ParameterizedType) javaType.getJavaType() ).getActualTypeArguments();
+					keyClass = ReflectHelper.getClass( typeArguments[0] );
+					valueClass = ReflectHelper.getClass( typeArguments[1] );
+					context = JAXBContext.newInstance( MapWrapper.class, keyClass, valueClass );
+				}
+				else {
+					keyClass = Object.class;
+					valueClass = Object.class;
+					context = JAXBContext.newInstance( MapWrapper.class );
+				}
+				final Unmarshaller unmarshaller = context.createUnmarshaller();
+				final MapWrapper mapWrapper = (MapWrapper) unmarshaller
+						.unmarshal( new StringReader( charSequence.toString() ) );
+				final List<Object> elements = mapWrapper.elements;
+				final Map<Object, Object> map = new LinkedHashMap<>( elements.size() >> 1 );
```

Comment:
This is not correct. Similar to `HashMap` and `HashSet`, we should not use `new LinkedHashMap<>(expectedSize)` for it would lead to rehashing when the size surpasses `expectedSize * 0.75`, thus defeating the purpose.
Our CollectionHelper class has a util method already: `CollectionHelper#linkedMapOfSize(expectedSize)`.
Such expected size misuse is so common. Not sure why JDK introduces some new method to avoid such confusion.